containers/Dockerfile.buildenv

```diff
@@ -309,3 +280,103 @@ RUN wget https://downloadmirror.intel.com/684899/sde-external-9.0.0-2021-11-07-l
+      pyyaml \
+    && pip2 --no-cache-dir install futures unittest2
+
+# Also install our minimum required CMake version so we can test it on CI.
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.12.4/cmake-3.12.4.tar.gz \
+    && tar -xzf cmake-3.12.4.tar.gz \
+    && cd cmake-3.12.4 \
+    && mkdir -p /opt/local/cmake/3.12.4 \
+    && ./bootstrap --prefix=/opt/local/cmake/3.12.4 \
+    && make $PARALLEL_MAKE_ARG\
+    && make install \
+    && cd .. && rm -rf cmake*
+
+# We build both Clang and GCC versions of Charm++ so that all our tests can
+# use the same build environment.
+WORKDIR /work/charm_7_0_0
+RUN ./build LIBS multicore-linux-x86_64 clang \
+      ${PARALLEL_MAKE_ARG} -g -O2 --build-shared\
+    && rm -rf /work/charm_7_0_0/doc /work/charm_7_0_0/examples
+
+# Remove the apt-get cache in order to reduce image size
+RUN apt-get -y clean
+
+WORKDIR /work
+
+
+# We don't inherit from ci because that has lots of unnecessary compilers that
+# we don't need and would only increase the size of the image. We inheret from
+# the remote image rather than the local dev because it is faster on release
+# CI to just pull the remote dev image and buiild demo rather than having to
+# build all of dev and demo combined.
+FROM sxscollaboration/spectre:dev AS demo
+
+# vim and emacs for editing files
+# Also ffmpeg for making movies with paraview output pngs
+# paraview needs curl
+RUN apt-get update -y \
+    && apt-get install -y vim emacs-nox ffmpeg curl
+
+# Install headless paraview so we can run pvserver in the container
+WORKDIR /work
+RUN wget -O paraview.tar.gz --no-check-certificate "https://www.paraview.org/paraview-downloads/download.php?submit=Download&version=v5.10&type=binary&os=Linux&downloadFile=ParaView-5.10.1-osmesa-MPI-Linux-Python3.9-x86_64.tar.gz" \
+  && tar -xzf paraview.tar.gz \
+  && rm paraview.tar.gz \
+  && mv ParaView-* /opt/paraview
+
+ENV PYTHONPATH "/usr/local/lib/python3.10/dist-packages"
+ENV PATH "/opt/paraview/bin:$PATH"
+
+WORKDIR /work
+
+# We copy the entire SpECTRE repo from the current context so that people can
+# edit and rebuild the pre-built executables if they want to experiment with
+# editing and building SpECTRE. The executables are chosen because they are the
+# ones in the "Hitchhikers's" tutorial for SpECTRE. More executables can be
+# added in later if we so desire.
+COPY . spectre/
+
+RUN mkdir spectre/build && cd spectre/build \
+  && cmake \
+    -D CMAKE_C_COMPILER=clang-10 \
+    -D CMAKE_CXX_COMPILER=clang++-10 \
+    -D CMAKE_Fortran_COMPILER=gfortran-9 \
+    -D OVERRIDE_ARCH=x86-64 \
+    -D CHARM_ROOT=/work/charm_7_0_0/multicore-linux-x86_64-gcc \
+    -D CMAKE_BUILD_TYPE=Release \
+    -D DEBUG_SYMBOLS=OFF \
+    -D BUILD_PYTHON_BINDINGS=ON \
+    -D PYTHON_EXECUTABLE=/usr/bin/python3 \
+    -D MEMORY_ALLOCATOR=SYSTEM \
+    .. \
+  && make $PARALLEL_MAKE_ARG ExportTimeDependentCoordinates3D \
+  && make $PARALLEL_MAKE_ARG EvolveScalarAdvectionKuzmin2D
```

Comment:
Also build all-pybindings I think, add bin/python to PYTHONPATH, and `pip install jupyterlab` for demos/tutorials.